Console/Command/BatchingOptimizeCommand.php

```diff
@@ -0,0 +1,366 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Console\Command;
+
+use Algolia\AlgoliaSearch\Exception\DiagnosticsException;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\ProductHelper;
+use Algolia\AlgoliaSearch\Service\AlgoliaConnector;
+use Algolia\AlgoliaSearch\Service\Product\IndexOptionsBuilder;
+use Algolia\AlgoliaSearch\Service\Product\RecordBuilder;
+use Algolia\AlgoliaSearch\Service\StoreNameFetcher;
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\Framework\App\Config\Storage\WriterInterface;
+use Magento\Framework\App\State;
+use Magento\Framework\Console\Cli;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class BatchingOptimizeCommand extends AbstractStoreCommand
+{
+    /**
+     * Recommended Max batch size
+     * https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/how-to/sending-records-in-batches/
+     */
+    const MAX_BATCH_SIZE = 10000000; //10MB
```

Comment:
I actually liked your earlier use of the [underscore numeric separator](https://php.watch/versions/7.4/underscore_numeric_separator). Any reason for abandoning it? I think rather than a comment we could just label it something like `MAX_BATCH_SIZE_IN_BYTES` for clarity.

Reply:
Oh ok my bad, I missread your earlier comment, I thought there was a compatibility issue with the underscore separators. Re-adding it.